### PR TITLE
[CMake] Fix install config files

### PR DESCRIPTION
**Pull request by fredroy**
More or less trial-and-error works to make the SofaPython3 plugin compilable out-of-tree (and obviously others one too)

Various stuff:
 - out-of-date cmake vars
 - need to reference old modules in the cmake.in (i dont really understand this issue)
 - remove completely SofaSimulationCore (was supposed to be deleted for v21.12) and add an error if used. I dont know if it is useful though. There were some f*ckey with the project/module called Sofa.SimulationCore but the folder SofaSimulationCore in any case.

I put fast-merge because this is quite urging but some cmake-specialized eyes are needed 👀 

By the way, with this branch, I can compile SofaPython3 out-of-tree on Windows/MSCV2022

______________________________________________________

By submitting this pull request, I acknowledge that  
**I have read, understand, and agree [SOFA Developer Certificate of Origin (DCO)](https://github.com/sofa-framework/sofa/blob/master/CONTRIBUTING.md#sofa-developer-certificate-of-origin-dco)**.
______________________________________________________

**Reviewers will merge this pull-request only if**  
- it builds with SUCCESS for all platforms on the CI.
- it does not generate new warnings.
- it does not generate new unit test failures.
- it does not generate new scene test failures.
- it does not break API compatibility.
- it is more than 1 week old (or has fast-merge label).

### CODE REVIEW

File: SofaGui/SofaGuiConfig.cmake.in [hunk 1]
```diff
@@ -10,12 +10,17 @@ find_package(Sofa.GUI.Component QUIET REQUIRED)
 find_package(Sofa.GUI.Common QUIET REQUIRED)
 find_package(Sofa.GUI.Batch QUIET REQUIRED)
 
+# keep references to the deprecated modules
+find_package(SofaGuiCommon QUIET REQUIRED)
+
 if(SOFAGUI_HAVE_SOFAGUIQT)
     find_package(Sofa.GUI.Qt QUIET REQUIRED)
+    find_package(SofaGuiQt QUIET REQUIRED)
```
Comment (alxbilger) [inline SofaGui/SofaGuiConfig.cmake.in:18]: This is quite annoying to keep the deprecated modules. This means that the compatibility is required, even in the SOFA code base.

**Review by alxbilger — APPROVED** [review]

File: SofaGui/SofaGuiConfig.cmake.in (hunk 1 above)
Comment (fredroy) [reply SofaGui/SofaGuiConfig.cmake.in:18]: Thats true.
I would say, lets keep it like that in order to fix the install (with compat) at least, and I will try to figure out something w/o the compat later

Comment (alxbilger) [reply SofaGui/SofaGuiConfig.cmake.in:18]: I agree

Comment (fredroy) [reply SofaGui/SofaGuiConfig.cmake.in:18]: Actually, after checking, SofaGui (and co) are not in the compat layer yet, as runSofa still needs it.  (I forgot about it earlier)

**Review by alxbilger — APPROVED** [review]

**Comment by fredroy** [comment]
> tested and approved on XUbuntu!

Same here (with manjaro up-to-date)

**Comment by guparan** [comment]
Thanks for fixing the SofaSimuationCore / Sofa.SimulationCore mess @fredroy :+1:

**Pull request merged**